networking_cisco/plugins/cisco/db/l3/l3_router_appliance_db.py

```diff
@@ -223,6 +223,11 @@ def create_router(self, context, router):
             # Ensure create spec is compliant with any HA
             ha_spec = self._ensure_create_ha_compliant(r, router_type)
         auto_schedule, share_host = self._ensure_router_scheduling_compliant(r)
+        driver = self._get_router_type_driver(context,
+                                              router_type_id)
+        if driver:
+            router_ctxt = driver_context.RouterContext(router)
+            driver.create_router_precommit(context, router_ctxt)
```

Comment:
It's a good question. I only added the callouts that are needed for our APIC L3 driver. I probably should add the callouts for the other ones, even though our driver won't have implementations for them. I'll update this patch to add callouts for everything defined by the API.